test/core-language/test.jam

```diff
@@ -1505,11 +1505,32 @@ check-equal shell : "value\n"   : [ SHELL $(c) ] ;
 check-equal shell : ""          : [ SHELL $(c) : no-output ] ;
 check-equal shell : "value\n" 0 : [ SHELL $(c) : exit-status ] ;
 check-equal shell : ""      0   : [ SHELL $(c) : no-output : exit-status ] ;
+check-equal shell : ""      0   : [ SHELL $(c) : no-output : exit-status : strip-eol ] ;
 check-equal command : "value\n"   : [ COMMAND $(c) ] ;
 check-equal command : ""          : [ COMMAND $(c) : no-output ] ;
 check-equal command : "value\n" 0 : [ COMMAND $(c) : exit-status ] ;
 check-equal command : ""      0   : [ COMMAND $(c) : no-output : exit-status ] ;
 
+# buffered output
+
+local expected = "When the shell output buffer splits on whitespace, the whitespace shouldn't be trimmed.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             end." ;
+local buffered = "echo \"$(expected)\"" ;
+if $(OS) = VMS { buffered = "PIPE WRITE SYS$OUTPUT \"$(expected)\"" ; }
```

Comment:
You need to add this line after 1518:

`if $(OS) = NT { buffered = "echo $(expected)" ; }`

That addition fixes the problem where your test does not pass under Windows.